src/main/java/cn/nukkit/utils/BinaryStream.java

```diff
@@ -473,6 +476,26 @@ public Item getSlot(int protocolId) {
             canDestroy[i] = this.getString();
         }
 
+        try {
+            if (nbt.length > 0) {
+                CompoundTag tag = Item.parseCompoundTag(nbt.clone());
+                int originalID = tag
+                        .getCompound("NukkitPetteriM1Edition")
+                        .getInt("OriginalID");
+                if (originalID != 0) {
+                    id = originalID;
```

Comment:
Maybe we should verify that this is some of the allowed item ids

Reply:
I added a simple check

Reply:
Should we check the `id` too?

Reply:
I think we can replace items in this range, and lower versions can be replaced with update blocks
In this way, even if the lower version players accidentally pick up the higher version items, they can also throw them away instead of getting stuck in the backpack
(Google Translate)

Reply:
I mean check that the `id` is some of the diamond items so no hacks could get for example air replaced with a netherite item by sending edited nbt

Reply:
nk’s inventory seems to be calculated by itself and will not directly trust the data package

Reply:
My time here is 9 pm, I will go to bed later, I will test and improve these things tomorrow

Reply:
> nk’s inventory seems to be calculated by itself and will not directly trust the data package

yeah but I have seen that failing enough times to say that some extra checks are not a bad idea

> My time here is 9 pm, I will go to bed later, I will test and improve these things tomorrow

ok

Reply:
> yeah but I have seen that failing enough times to say that some extra checks are not a bad idea

The previous error was because I set the wrong nbt, and nk could not judge the item correctly   QAQ